bindata/network/openshift-sdn/sdn.yaml

```diff
@@ -174,6 +174,42 @@ spec:
             command: ["test", "-f", "/etc/cni/net.d/80-openshift-network.conf"]
           initialDelaySeconds: 5
           periodSeconds: 5
+      - name: kube-rbac-proxy
+        image: {{.KubeRBACProxyImage}}
+      - command:
+        - /bin/bash
+        - -c
+        - |
+          #!/bin/bash
+          set -euo pipefail
+          TLS_PK=/etc/pki/tls/metrics-certs/tls.key
+          TLS_CERT=/etc/pki/tls/metrics-certs/tls.crt
+
+          # As the secret mount is optional we must wait for the files to be present.
+          while [[ ! -f ${TLS_PK} ||  ! -f ${TLS_CERT} ]] ; do
+            echo $(date -Iseconds) WARN: Waiting for sdn-metrcs-certs to be created and mounted. This is expected during the installation.
```

Comment:
@juanluisvaladas two things; how about INFO instead of WARN if this is expected? Second, should we ever exit the loop, or just do we just wait for something else to kill the pod?

Reply:
> Or, perhaps INFO for the first 5 minutes, then WARN? Especially if the secrets directory is optional and might not be mounted the first time the container is run.

I like this, but the container creation time isn't really correlated with the certificate creation time.

I'm not quite sure how long it actually takes to create, but the whole instalation takes 30 so how about this:
1- Check if the files are there.
2- If they aren't report it once,
3- Check periodically and don't report anything at all until service creation timestamp + 25 minutes of grace period (if the whole installation takes 30, 25 seems very generous to me). After 25 minutes report it as an error rather than a warning.
4- After a longer period, maybe an hour or something in that line maybe make it a fatal so that there is a crashloop back off and it triggers alarms and the CNO to report issues

Reply:
Didn't want to wait any further because this may be having an actual impact so I implemented this grace period anyway. Let me know if you don't like it @dcbw